### PR TITLE
Audit improvements

### PR DESCRIPTION
**Pull request by Gwydce**

### CODE REVIEW

File: contracts/AaveStrategy.sol [hunk 1]
```diff
@@ -291,12 +301,28 @@ contract AaveStrategy is CommonContract {
...
+            // E.g want is USDT with  only 6 decimals:
+            // _maticToWantRatio = 1_522_650_000 (1.52265 USDT/MATIC)
+            // balance = 1e18 (1.0 MATIC)
+            // tokenDiffPrecision = 1e12 (1e18 MATIC decimals / 1e6 USDT decimals)
+            // expected = 1522650 (1e18 * 1_522_650_000 / 1e9 / 1e12) [1.52 in USDT decimals]
+
+            uint tokenDiffPrecision = ERC20(wmatic).decimals().div(
+                ERC20(want).decimals()
+            );
+            uint expected = balance.mul(_maticToWantRatio).div(1e9).div(tokenDiffPrecision);
```
Comment (maxsam4) [inline contracts/AaveStrategy.sol:322]: You can not have decimal values in `uint`. Therefore, multiply the 9 decimal precision before doing any divisions.
```suggestion
            uint tokenDiffPrecision = ERC20(wmatic).decimals().mul(1e9).div(
                ERC20(want).decimals()
            );
            uint expected = balance.mul(_maticToWantRatio).div(tokenDiffPrecision);
```

File: contracts/AaveStrategy.sol [hunk 2]
```diff
@@ -13,6 +13,7 @@ contract AaveStrategy is CommonContract {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint;
+    using SafeMath for uint8;
```
Comment (maxsam4) [inline contracts/AaveStrategy.sol:16]: FYI solidity 0.8 comes with inbuilt safemath. you do not need to explicitly use any SafeMath library.

File: contracts/AaveStrategy.sol (hunk 1 above)
Comment (Gwydce) [reply contracts/AaveStrategy.sol:322]: Checking this again... the tokenDiffPrecision should be the real precision, not the number of decimals, my bad. This should be the correct version, what do you think?:
```solidity
            uint tokenDiffPrecision = (10 ** ERC20(wmatic).decimals()).div(
                10 ** ERC20(want).decimals()
            ).mul(1e9);
            uint expected = balance.mul(_maticToWantRatio).div(tokenDiffPrecision);
```

Comment (maxsam4) [reply contracts/AaveStrategy.sol]: Seems fine, just make sure to use the same method when calculating `_maticToWantRatio`

**Pull request merged**